aten/src/THC/THCTensorCopy.cu

```diff
@@ -8,12 +8,23 @@
 template <typename TypeDst, typename TypeSrc>
 struct CopyOp {
   __device__ __forceinline__ void operator()(TypeDst* dst, TypeSrc* src) {
-#if __CUDA_ARCH__ >= 350 && defined(TH_REAL_IS_BOOL)
+#if __CUDA_ARCH__ >= 350
     *dst = ScalarConvert<TypeSrc, TypeDst>::to(__ldg(src));
 #else
     *dst = ScalarConvert<TypeSrc, TypeDst>::to(*src);
 #endif
   }
+
+  __device__ __forceinline__ void operator()(bool* dst, bool* src) {
+    *dst = ScalarConvert<TypeSrc, TypeDst>::to(*src);
+  }
+};
+
+template <>
+struct CopyOp <bool, bool> {
```

Comment:
I wouldn't fully specialize this unless you rewrite the above to only have one template parameter.  You can just partially specialize this.

Reply:
The reason being the above won't actually work if someone was to instantiate <non_bool, bool>.